### ISSUE TITLE
mutation want bind  to ent generated schema struct

### ISSUE THREAD

**Issue by godcong** [ent/ent#1474]
- [X] The issue is present in the latest release.
- [X] I have searched the [issues](https://github.com/facebook/ent/issues) of this repository and believe that this is not a duplicate.

## Current Behavior 😯

When we update some data through http, we usually use the struct generated by the schema.
However, when assigning values, we can only use the Set method to set them one by one.

## Expected Behavior 🤔

Is it possible to add binding relationship of struct to Mutation or Create, Update/UpdateOne to update it?
Of course, we sometimes don’t want to update all fields
We can add several key options to filter this condition, like: Expect，Only
## Steps to Reproduce 🕹

Steps:

```go
 //generate from schema
 type SomeTable struct{
    String1 string
    Int1 int
     //xxx data
 }
 
//this will only update string1 data
client.SomeTable.Create().Bind(someTable, Only(String1))
//or
client.SomeTable.Create().Mutation().Bind(someTable, Only(String1))

//this will update someTable expect string1 data
client.SomeTable.Create().Bind(someTable, Expect(String1))
//or
client.SomeTable.Create().Mutation()Bind(someTable, Expect(String1))
```
I try to use `Template` to achieve this, but it seems that I have no way to get all the Fields.
`Fields` of `Mutation` can only get non-null values.

## Your Environment 🌎

| Tech        | Version |
| ----------- | ------- |
| Go          | 1.16.3  |
| Ent         | 0.8.0   |
| Database    | Mysql   |
| Driver      | https://github.com/go-sql-driver/mysql        |

**Comment by wahello** [comment]
Maybe you want is set members' value in one batch, not just set member value one by one, like 
`
client.User.Create().
SetAge(22).
SetGender("m").
SetFirstName("tom")

`

Is that's right?

**Comment by a8m** [comment]
Hey @godcong and it's great to see you here again 😃 

I think you're looking for something like the [mutation-input](https://entgo.io/docs/tutorial-todo-gql-mutation-input) tutorial, or [this example](https://entgo.io/docs/faq#how-to-create-an-entity-from-a-struct-t) from our FAQ page. 

TL;DR - you can generate input types (for GraphQL or REST) and use only one method for setting the values on the input object automatically. For example:

```go
client.Todo.Create().SetInput(input).Save(ctx)
```

Please let me know if you need any help with implementing this. I think this helper is going to be part of the official `ent/contrib` packages really soon.

**Comment by godcong** [comment]
@a8m I tried it and it works
Now I share my template with everyone.
Some methods of Softdelete are added inside

```
{{/* The line below tells Intellij/GoLand to enable the autocompletion based on the *gen.Graph type. */}}
{{/* gotype: entgo.io/ent/entc/gen.Graph */}}

{{ define "template" }}
    {{/* ... */}}
{{ end }}

{{ define "input" }}

    {{/* Add the base header for the generated file */}}
    {{ $pkg := base $.Config.Package }}
    {{ template "header" $ }}

    {{ range $n := $.Nodes }}
        {{ $creater := $n.CreateName }}
        {{ $createReceiver := receiver $creater }}

        func ({{ $createReceiver }} *{{ $creater }}) Set{{ $n.Name }}(input *{{ $n.Name }}) *{{ $creater }} {
        {{- range $f := $n.Fields }}
            {{- if ne $f.StructField "DeletedUnix"}}
                {{- $setter := print "Set" $f.StructField }}
                {{ $createReceiver }}.{{ $setter }}(input.{{ $f.StructField }})
            {{- end}}
        {{- end }}
        return {{ $createReceiver }}
        }

        {{- range $f := $n.Fields }}
            {{- if eq $f.StructField "DeletedUnix"}}
                func ({{ $createReceiver }} *{{ $creater }}) SoftDelete() *{{ $creater }} {
                {{- $setter := print "Set" $f.StructField }}
                {{ $createReceiver }}.{{ $setter }}(time.Now().Unix())
                return {{ $createReceiver }}
                }
            {{- end}}
        {{- end }}

        {{ $updater := $n.UpdateName }}
        {{ $updateReceiver := receiver $updater }}

        func ({{ $updateReceiver }} *{{ $updater }}) Set{{ $n.Name }}(input *{{ $n.Name }}) *{{ $updater }} {
        {{- range $f := $n.Fields }}
            {{- if ne $f.StructField "DeletedUnix"}}
                {{- $setter := print "Set" $f.StructField }}
                {{ $updateReceiver }}.{{ $setter }}(input.{{ $f.StructField }})
            {{- end}}
        {{- end }}
        return {{ $updateReceiver }}
        }

        {{ $updateOner := $n.UpdateOneName }}
        {{ $updateOneReceiver := receiver $updateOner }}

        func ({{ $updateOneReceiver }} *{{ $updateOner }}) Set{{ $n.Name }}(input *{{ $n.Name }}) *{{ $updateOner }} {
        {{- range $f := $n.Fields }}
            {{- if ne $f.StructField "DeletedUnix"}}
                {{- $setter := print "Set" $f.StructField }}
                {{ $updateOneReceiver }}.{{ $setter }}(input.{{ $f.StructField }})
            {{- end}}
        {{- end }}
        return {{ $updateOneReceiver }}
        }

        {{ $querier := $n.QueryName }}
        {{ $queryReceiver := receiver $querier }}
        {{- range $f := $n.Fields }}
            {{- if eq $f.StructField "DeletedUnix"}}
                func ({{ $queryReceiver }} *{{ $querier }}) SoftDelete() *{{ $querier }} {
                {{- $setter := print  $f.StructField "IsNil"}}
                return {{ $queryReceiver }}.Where({{$n.Package}}.{{ $setter }}())
                }
            {{- end}}
        {{- end }}

    {{ end }}
{{ end }}
```

**Comment by a8m** [comment]
Thanks for sharing this @godcong 🙏
